tests/unit/delivery-options/components/pickup/Pickup.spec.js

```diff
@@ -0,0 +1,60 @@
+import * as CONFIG from '@/data/keys/configKeys';
+import Pickup from '@/delivery-options/components/Pickup/Pickup';
+import { configBus } from '@/delivery-options/config/configBus';
+import { createPickupChoices } from '@/delivery-options/data/pickup/createPickupChoices';
+import { fetchAllCarriers } from '@/delivery-options/data/carriers/fetchAllCarriers';
+import { mockVue } from '../../mockVue';
+import { mount } from '@vue/test-utils';
+import { mockDeliveryOptions } from '@Tests/unit/delivery-options/mockDeliveryOptions';
+
+describe('Pickup.vue', () => {
+  let localVue;
+  let wrapper;
+
+  beforeAll(async() => {
+    localVue = mockVue({
+      [CONFIG.KEY]: {
+        [CONFIG.FEATURE_MAX_PAGE_ITEMS]: 5,
+      },
+    });
+
+    await fetchAllCarriers();
+    configBus.setAdvancedCarrierData();
+
+    const choices = await createPickupChoices();
+    wrapper = mockDeliveryOptions(Pickup, {
+      localVue,
+      propsData: {
+        data: {
+          choices,
+        },
+      },
+    });
+  });
+
+  it('shows map view by default', () => {
+    expect(wrapper.findByTestId('view--map').element).toBeVisible();
+    expect(wrapper.findByTestId('view--list').exists()).toBeFalsy();
+  });
+
+  it('can toggle between list and map view', async() => {
+    expect.assertions(2);
+    await wrapper.findByTestId('button--list').element.click();
+
+    expect(wrapper.findByTestId('view--map').exists()).toBeFalsy();
+    expect(wrapper.findByTestId('view--list').element).toBeVisible();
+  });
+
+  it('has functional pagination', async() => {
+    expect.assertions(3);
+    await wrapper.findByTestId('button--list').element.click();
+
+    const loadMore = 'button--load-more';
+    const pickupLocation = 'tr[data-test-id^="pickupLocation"]';
```

Comment:
Vanwaar de tr? Je data-test-id zou toch uniek moeten zijn?

Reply:
Dat is wel zo ja, maar deze elementen zijn `pickupLocation--<option.name>` maar de inputs daarbinnen zijn `pickupLocation--<option.name>__input`. Dus het is dit of ik moet een manier maken om regex te gebruiken hier